packages/core/src/driver.ts

```diff
@@ -250,6 +343,80 @@ class Driver {
     this._afterConstruction()
   }
 
+  /**
+   * The bookmark managed used by {@link Driver.executeQuery}
+   *
+   * @type {BookmarkManager}
+   * @returns {BookmarkManager}
+   */
+  get queryBookmarkManager (): BookmarkManager {
+    return this._queryBookmarkManager
+  }
+
+  /**
+   * Executes a query in a retriable context and returns a {@link EagerResult}.
+   *
+   * This method is a shortcut for a transaction function
+   *
+   * @example
+   * // Run a simple write query
+   * const { keys, records, summary } = await driver.executeQuery('CREATE (p:Person{ name: $name }) RETURN p', { name: 'Person1'})
+   *
+   * @example
+   * // Run a read query
+   * const { keys, records, summary } = await driver.executeQuery(
+   *                  'MATCH (p:Person{ name: $name }) RETURN p',
+   *                  { name: 'Person1'},
+   *                  { routing: neo4j.routing.READERS})
+   *
+   * @example
+   * // this lines
+   * const transformedResult = await driver.executeQuery(
+   *       "<QUERY>",
+   *       <PARAMETERS>,
+   *       QueryConfig {
+   *           routing: neo4j.routing.WRITERS,
+   *           resultTransformer: transformer,
+   *           database: "<DATABASE>",
+   *           impersonatedUser: "<USER>",
+   *           bookmarkManager: bookmarkManager
+   *       }
+   *   )
+   * // are equivalent to this ones
+   * const session = driver.session({
+   *            database: "<DATABASE>",
+   *            impersonatedUser: "<USER>",
+   *            bookmarkManager: bookmarkManager
+   * })
+   *
+   * try {
+   *    const transformedResult = await session.executeWrite(tx => {
+   *        const result = tx.run("<QUERY>", <PARAMETERS>)
+   *        return transformer(result)
+   *    })
+   * } finally {
+   *    await session.close()
+   * }
+   *
+   * @public
+   * @param {string| {text:string, parameters?: object}} query - Cypher query to execute
+   * @param {Object} parameters - Map with parameters to use in query
+   * @param {QueryConfig<T>} config - The query configuration
+   * @returns {Promise<T>}
+   */
+  async executeQuery<T> (query: Query, parameters?: any, config: QueryConfig<Dict, T> = {}): Promise<T> {
+    const bookmarkManager = config.bookmarkManager === null ? undefined : config.bookmarkManager ?? this.queryBookmarkManager
+    const resultTransformer: ResultTransformer<T> = (config.resultTransformer ?? createEagerResultFromResult) as unknown as ResultTransformer<T>
```

Comment:
Because of the `Dict` thing, the typing was not matching 100%. In this kind scenario, you only can force cast if you cast to unknown first<sup>1</sup>. But it is already fixed.

<sup><sup>1</sup>you can actually cast, but the linter just screams in your face :D </sup>